### PR TITLE
Depend on detached kotlin-native-shared 

### PR DESCRIPTION
**Pull request by ilmat192**

### CODE REVIEW

File: buildSrc/plugins/build.gradle [hunk 1]
```diff
@@ -37,9 +38,7 @@ dependencies {
     compile "org.jetbrains.kotlin:kotlin-stdlib:$buildKotlinVersion"
     compile "org.jetbrains.kotlin:kotlin-reflect:$buildKotlinVersion"
     compile group: 'com.ullink.slack', name: 'simpleslackapi', version: '0.6.0'
-    // An artifact from the included build 'shared' cannot be used here due to https://github.com/gradle/gradle/issues/3768
-    // TODO: Remove this hack when the bug is fixed
-    compile project(':shared')
```
Comment (vvlevchenko) [inline buildSrc/plugins/build.gradle]: 👍

File: cmd/run_konan [hunk 2]
```diff
@@ -85,13 +85,14 @@ KOTLIN_SCRIPT_RUNTIME_JAR="${KONAN_HOME}/konan/lib/kotlin-script-runtime.jar"
 STUB_GENERATOR_JAR="${KONAN_HOME}/konan/lib/StubGenerator.jar"
 INTEROP_INDEXER_JAR="${KONAN_HOME}/konan/lib/Indexer.jar"
 INTEROP_JAR="${KONAN_HOME}/konan/lib/Runtime.jar"
-SHARED_JAR="${KONAN_HOME}/konan/lib/shared.jar"
+SHARED_JAR="${KONAN_HOME}/konan/lib/kotlin-native-shared.jar"
```
Comment (olonho) [inline cmd/run_konan]: Would be great to combine all JARs we can into the single one eventually, to avoid changes here.

File: settings.gradle [hunk 3]
```diff
@@ -32,7 +32,7 @@ include ':utilities'
 //include ':tools:benchmarksAnalyzer'
 include ':platformLibs'
 
-includeBuild 'shared'
+includeBuild 'version'
```
Comment (olonho) [inline settings.gradle]: How to make composite build with shared?

Comment (ilmat192) [reply settings.gradle]: Added the composite build. Now it's possible to specify path to a local shared repo using the corresponding property:
```
./gradlew dist -PsharedProjectPath=/path/to/shared
```

**Pull request merged**